apps/vaporgui/MainForm.cpp

```diff
@@ -1929,7 +1929,12 @@ void MainForm::_performSessionAutoSave()
 
     if (_eventsSinceLastSave >= eventCountForAutoSave) {
         string autoSaveFile = sParams->GetAutoSaveSessionFile();
-        _paramsMgr->SaveToFile(autoSaveFile);
+        int    rc = _paramsMgr->SaveToFile(autoSaveFile);
+        if (rc == -1) {
```

Comment:
The error handling here seems redundant and more complex than needed. ParamsMgr::SaveToFile() already distinguishes between a file write vs file access error and prints an appropriate message for each case. The message generated here replicates the message and doesn't add any additional information. Why not simply leave SaveToFile() unchanged (returning a single error code), and not have the code here detect between the two error conditions and generate what is essentially the same message?

Reply:
MSG_ERR will include any messages that have been reported by MyBase::SetErrMsg

Reply:
Got it.  Fixed.